proposals/0180-leader-schedule-migration.md

```diff
@@ -0,0 +1,91 @@
+---
+simd: '0180'
+title: Leader Schedule Migration
+authors: Justin Starry (Anza)
+category: Standard
+type: Core
+status: Draft
+created: 2024-10-03
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+The epoch leader schedule for block production will be migrated from using
+validator identity addresses to using vote account addresses. The expected
+block signer for a given slot will be determined by the vote account's 
+designated validator identity.
+
+## Motivation
+
+Using validator identity addresses in the leader schedule means there is no
+straightforward way to map a block producer to a particular vote account and its
+delegated stake. This is becuase the same validator identity could be designated
+by multiple vote accounts. By migrating to the vote account address, we know
+exactly what delegated stake led to a validator's leader schedule slot
+allocation. This will make certain protocol improvements much easier to design
+like how to distribute block rewards and how to slash validators that produce
+duplicate blocks.
+
+## New Terminology
+
+NA
+
+## Detailed Design
+
+### Leader Schedule Generation
+
+When generating the leader schedule at epoch boundaries, rather than
+accumulating all stake by the node id, stake should be accumulated according to
+vote pubkey. Then use the existing stake weighted randomized leader schedule
+generation using vote pubkeys and their delegated stake rather than node id
+pubkeys and the accumulated delegated stake across (potentially more than one)
+vote accounts. As before, only valid and initialized vote accounts should be
```

Comment:
do we include any of the old vote state versions as "valid"? are validity criteria documented somewhere that we can link out to?

Reply:
I added a section for this.. we don't explicitly check vote state version for the initialization check. We just check the size of the account and whether there are some non-zero bytes. Less than ideal for sure but out of scope for this SIMD.